playground/backend/internal/executors/executor_builder_test.go

```diff
@@ -0,0 +1,210 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executors
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestNewExecutorBuilder(t *testing.T) {
+	tests := []struct {
+		name string
+		want *ExecutorBuilder
+	}{
+		{
+			name: "Get executor builder",
+			want: &ExecutorBuilder{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewExecutorBuilder(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewExecutorBuilder() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestExecutorBuilder_WithCompiler(t *testing.T) {
+	type fields struct {
+		actions []handler
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   *CompileBuilder
+	}{
+		{
+			name:   "Get CompileBuilder with empty actions",
+			fields: fields{actions: []handler{}},
+			want:   &CompileBuilder{ExecutorBuilder{actions: []handler{}}},
```

Comment:
Can we add cases when `actions` are not empty for all these tests?

Reply:
Done